src/gui/rss/automatedrssdownloader.cpp

```diff
@@ -521,12 +522,36 @@ void AutomatedRssDownloader::handleFeedCheckStateChange(QListWidgetItem *feed_it
 void AutomatedRssDownloader::updateMatchingArticles()
 {
     ui->treeMatchingArticles->clear();
+    saveEditedRule();
+    m_updateMatchingArticlesDeferrals.ref();
+    QTimer::singleShot(1000, this, SLOT(deferredUpdateMatchingArticles()));
```

Comment:
This will create a multitude of temporary QTimers. If I read it correctly it will create one for each keystroke.
Instead do this. In the class constructor create a member Timer. Set its property to singleshot and its interval to 1000ms.
Then in this function/slot, just call start() on the timer. If it is already running it will be restarted.